test/Mono.Linker.Tests.Cases/Reflection/RunClassConstructorUsedViaReflection.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Reflection;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+using System.Runtime.CompilerServices;
+
+namespace Mono.Linker.Tests.Cases.Reflection
+{
+	[SetupCSharpCompilerToUse ("csc")]
+	public class RunClassConstructorUsedViaReflection
+	{
+		public static void Main ()
+		{
+			TestRunClassConstructor ();
+			TestNonKeptStaticConstructor ();
+			TestNull ();
+			TestDataFlowType ();
+			TestIfElse (1);
+		}
+
+		[Kept]
+		static void TestRunClassConstructor ()
+		{
+			RuntimeHelpers.RunClassConstructor (typeof (OnlyUsedViaReflection).TypeHandle);
+		}
+
+		[Kept]
+		static void TestNonKeptStaticConstructor ()
+		{
+			var a = new NonKeptStaticConstructorClass();
+		}
+
+		[Kept]
+		static void TestNull ()
+		{
+			Type type = null;
+			RuntimeHelpers.RunClassConstructor (type.TypeHandle);
```

Comment:
This should also have `RecognizedReflectionAccessPattern` attribute to validate that it doesn't issue a warning.